src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,217 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement = connection.prepareStatement(updateQuery)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update a car: " + car, e);
+        }
+        removeAllDriversFromCar(car.getId());
+        insertDriversForCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        boolean isCarDeleted;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStatement = connection.prepareStatement(deleteQuery)) {
+            deleteStatement.setLong(1, id);
+            isCarDeleted = deleteStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete a car with id: " + id, e);
+        }
+        removeAllDriversFromCar(id);
+        return isCarDeleted;
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverQuery = "SELECT c.model as car_model, c.id as car_id, "
+                                     + "manufacturer_id, m.name as manufacturer_name, "
+                                     + "m.country as manufacturer_country "
+                                     + "FROM cars c "
+                                     + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                                     + "JOIN cars_drivers cd ON cd.car_id = c.id "
+                                     + "WHERE c.is_deleted = FALSE AND cd.driver_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllByDriverStatement =
+                        connection.prepareStatement(getAllByDriverQuery)) {
+            getAllByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllByDriverStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(parseCarFromResultSet(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get cars by a driver ID: " + driverId, e);
+        }
+    }
+
+    private Car parseCarFromResultSet(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject("car_id", Long.class));
+        car.setModel(resultSet.getString("car_model"));
+        car.setManufacturer(parseManufacturer(resultSet));
+        car.setDrivers(getDriversByCar(car.getId()));
```

Comment:
you definitely shouldn't put this getDriversByCar call here as it rises a problem of nested connections that is described in common mistakes